src/KeyringController.ts

```diff
@@ -785,21 +960,22 @@ class KeyringController extends EventEmitter {
       const accounts = candidate[1].map(normalizeAddress);
       return accounts.includes(hexed);
     });
-    if (winners && winners.length > 0) {
+
+    if (winners?.length && winners?.[0]?.length) {
       return winners[0][0];
     }
 
     // Adding more info to the error
     let errorInfo = '';
     if (!address) {
       errorInfo = 'The address passed in is invalid/empty';
-    } else if (!candidates || !candidates.length) {
+    } else if (!candidates?.length) {
```

Comment:
It doesn't seem that you used `?.` when referring to `candidates` further up this method, so do you need the `?.` here?

Reply:
No, removing it

Reply:
Removed at [f609772](https://github.com/MetaMask/KeyringController/pull/202/commits/f609772065886c279c181bb532dfe5cdb4c94751)